tests/PHPStan/TranslationFunctionRuleTest/InClass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace SilverStripe\Standards\Tests\PHPStan\TranslationFunctionRuleTest;
+
+use SilverStripe\i18n\i18n;
+
+class InClass
+{
+    public const MY_ENTITY = 'entity';
+
+    public function myMethod()
+    {
+        _t('abc');
+        _t(InClass::class);
+        _t(InClass::class . 'somekey');
+        _t(InClass::MY_ENTITY);
+        _t(__CLASS__ . InClass::MY_ENTITY);
+        $variable = 'somethingwrong';
+        _t($variable);
+        i18n::_t('nodot');
+        $callableString = '_t';
+        $callableString('nodot');
```

Comment:
If you're going to this degree of detection, you may as well also include call_user_func() and call_user_func_array()

Reply:
PHPStan doesn't treat those the same way, so that would require adding a bunch of code for an extreme edge case. I'd rather not do that, especially given text collector won't identify those anyway.